lib/libzfs_core/libzfs_core.c

```diff
@@ -645,6 +658,90 @@ lzc_send_resume(const char *snapname, const char *from, int fd,
 	    resumeoff, NULL));
 }
 
+int
```

Comment:
this can and will overflow
```
#include <stdio.h>
int
lzc_get_pipe_max()
{
        unsigned long max_psize = 65536;
#ifdef __linux__
        FILE *procf = fopen("/proc/sys/fs/pipe-max-size", "re");

        if (procf != NULL) {
                if (fscanf(procf, "%lu", &max_psize) <= 0) {
                        // safe default
                        max_psize = 65536;
                }
                fclose(procf);
        }
#endif
        return (max_psize);
}

int main() {
        int ret = lzc_get_pipe_max();
        printf("%d\n", ret);
}
nabijaczleweli@tarta:~/uwu$ cc max.c -omax
nabijaczleweli@tarta:~/uwu$ echo 2147483649 | sudo tee /proc/sys/fs/pipe-max-size
2147483649
nabijaczleweli@tarta:~/uwu$ ./max
-2147483648
```

why not return an unsigned long?

Reply:
Very good point; I overlooked that it was using unsigned longs when I factored out the logic. Corrected.